pkg/gofr/datasource/kv-store/badger/badger_test.go

```diff
@@ -0,0 +1,65 @@
+package badger
+
+import (
+	"bytes"
+	"context"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	"os"
+	"testing"
+)
+
+func setupDB(t *testing.T) *client {
+	t.Helper()
+	cl := New(Configs{DirPath: "test_badger"})
+
+	var logs []byte
+
+	ctrl := gomock.NewController(t)
+
+	cl.UseLogger(NewMockLogger(DEBUG, bytes.NewBuffer(logs)))
+	cl.UseMetrics(NewMockMetrics(ctrl))
+	cl.Connect()
+
+	return cl
+}
+
+func Test_ClientSet(t *testing.T) {
+	cl := setupDB(t)
+	defer os.RemoveAll("test_badger")
```

Comment:
Move this to a t.Cleanup(func () {}) inside setupDB this way you won't need this defer in each tests

https://pkg.go.dev/testing#T.Cleanup

It's the purpose and the magic that t.Cleanup brought